test/unit/taxonomy/world_taxonomy_test.rb

```diff
@@ -7,14 +7,15 @@ def subject
     Taxonomy::WorldTaxonomy.new
   end
 
-  test "#world_taxons" do
+  test "#sort world_taxons alphabetically when children == 0, then alphabetically on world_taxons with children" do
     redis_cache_has_world_taxons(
       [
-        build(:taxon_hash, content_id: 'content_id_1'),
-        build(:taxon_hash, content_id: 'content_id_2'),
-        build(:taxon_hash, content_id: 'content_id_3')
+        build(:taxon_hash, title: 'France', children: [child_taxon]),
+        build(:taxon_hash, title: 'Australia', children: [child_taxon, child_taxon]),
+        build(:taxon_hash, title: 'Vietnam'),
+        build(:taxon_hash, title: 'Germany')
```

Comment:
I'd use more realistic data here, like a title of "News and Events", just to help people in the future make sense of the tests.